### PR TITLE
feat: add install flag

### PR DESCRIPTION
**Pull request by cardoso**
I had a nightmarish xtask to do the packaging and then I saw pluginify. It did everything better. I deleted the dreadful xtask a second later. Thanks for the plugin @itowlson !

The only thing I missed was an install flag. I'd also like to move almost everything from main.rs to a lib.rs in a separate PR if it's okay. It may be useful as a library for xtasks, build scripts, and other plugins.

### CODE REVIEW

**Comment by cardoso** [comment]
Reverted the main.rs -> lib.rs move to make the diff readable, but there's still cargo fmt happening... I can undo that as well, but otherwise the real change is only the install flag now.

**Review by cardoso — COMMENTED** [review]
revert: cargo fmt

File: README.md [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
+* [Spin](https://developer.fermyon.com/spin/install)
+
 ## Installation
 
-```
-cargo build --release
-./target/release/pluginify
-spin plugins install --file pluginify.json --yes
+```bash
+cargo run -r -- --install
```
Comment (itowlson) [inline README.md:13]: Ooooooooooooooooooh

File: src/spin.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+use std::process::Command;
+
+pub fn plugin_install_file(file: PathBuf) -> Result<(), std::io::Error> {
+    let spin = std::env::var("SPIN_BIN_PATH").unwrap_or("spin".into());
+
+    Command::new(spin)
+        .arg("plugin")
+        .arg("install")
+        .arg("--file")
+        .arg(file)
```
Comment (itowlson) [inline src/spin.rs:11]: Might want `--yes` here, otherwise Spin will prompt, which seems unnecessary in this situation.

**Review by itowlson — COMMENTED** [review]
This is super - I kept thinking "I wish this would install the plugin too" - I particularly love that you can now install pluginify itself by doing a `cargo run`.

However, I'm a bit unsure if this PR is complete - it seems like the `--install` flag is not actually hooked up in the clap args?

**Comment by itowlson** [comment]
Looking at my notifications I think this is WIP and I've jumped in too early with my comments - sorry about that.

**Comment by cardoso** [comment]
Oh my bad @itowlson ! The PR was too big, so I cleaned up to send in other PRs and went too far, but the flag is back now :)

File: src/spin.rs (hunk 2 above)
Comment (cardoso) [reply src/spin.rs:11]: Makes sense! Added

File: src/spin.rs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+use std::process::Command;
+
+pub fn plugin_install_file(file: PathBuf) -> Result<(), std::io::Error> {
+    let spin = std::env::var("SPIN_BIN_PATH").unwrap_or("spin".into());
+
+    Command::new(spin)
+        .arg("plugin")
+        .arg("install")
+        .arg("--file")
+        .arg("--yes")
```
Comment (itowlson) [inline src/spin.rs]: Not sure if the ordering will work - won't this need to be `--file file --yes`?

**Review by itowlson — APPROVED** [review]
LGTM - thank you!  Is this ready to merge from your point of view?  I'm happy, I just don't want to disappear the PR if you're still making changes!

**Comment by cardoso** [comment]
All good now @itowlson !  Thanks for the quick review

**Pull request merged**